pennylane/transforms/dynamic_one_shot.py

```diff
@@ -228,7 +233,7 @@ def measurement_with_no_shots(measurement):
             else np.nan
         )
 
-    interface = qml.math.get_deep_interface(circuit.data)
+    interface = interface or qml.math.get_deep_interface(circuit.data)
```

Comment:
Inside inner execution, `circuit.data` has already been converted to numpy. The original interface was passed into the transform via the added keyword argument and used here.